js/fcoin.js

```diff
@@ -0,0 +1,572 @@
+            let markets = res['data'];
+            for (let p = 0; p < markets.length; p++) {
+                let market = markets[p];
+                let id = market['name'];
+                let baseId = market['base_currency'].toUpperCase ();
+                let quoteId = market['quote_currency'].toUpperCase ();
+                let base = this.commonCurrencyCode (baseId);
+                let quote = this.commonCurrencyCode (quoteId);
+                let symbol = base + '/' + quote;
+                let precision = {
+                    'price': market['price_decimal'],
+                    'amount': market['amount_decimal'],
+                };
+                let limits = {
+                    'price': {
+                        'min': Math.pow (10, -precision['price']),
+                        'max': Math.pow (10, precision['price']),
+                    },
+                };
+                if (symbol in this.limits) {
+                    limits['amount'] = {
+                        'min': this.safeFloat (this.limits[symbol]['amount'], 'min'),
+                        'max': this.safeFloat (this.limits[symbol]['amount'], 'max'),
+                    };
+                } else {
+                    limits['amount'] = {
+                        'min': 0.01,
+                        'max': 100000,
+                    };
+                }
+                result.push ({
+                    'id': id,
+                    'symbol': symbol,
+                    'base': base,
+                    'quote': quote,
+                    'baseId': baseId,
+                    'quoteId': quoteId,
+                    'active': true,
+                    'precision': precision,
+                    'limits': limits,
+                    'info': market,
+                });
+            }
+        }
+        return result;
+    }
+
+    async fetchBalance (params = {}) {
+        await this.loadMarkets ();
+        let res = await this.privateGetAccountsBalance ();
+        let result = { 'info': res };
+        let balances = res['data'];
+        for (let i = 0; i < balances.length; i++) {
+            let balance = balances[i];
+            let currency = balance['currency'];
+            let uppercase = currency.toUpperCase ();
+            uppercase = this.commonCurrencyCode (uppercase);
+            let account = this.account ();
+            account['free'] = parseFloat (balance['available']);
+            account['total'] = parseFloat (balance['balance']);
+            account['used'] = parseFloat (balance['frozen']);
+            result[uppercase] = account;
+        }
+        return this.parseBalance (result);
+    }
+
+    convertOrderBook (bidasks) {
+        let newbidasks = [];
+        let length = bidasks.length / 2;
+        for (let i = 0; i < length; i++) {
+            newbidasks.push (bidasks.slice (0, 2));
+            bidasks = bidasks.slice (2);
+        }
+        return newbidasks;
```

Comment:
This method looks very suspicios, it just cannot work this way properly. You're returning a plain array `[]` with two elements from here, and later in https://github.com/ccxt/ccxt/pull/3225/files#diff-36563d247b2186c9c67172fba66dcdf0R263 you're trying to accessing the keys `bids` and `asks` in the plain array that only has indices 0 and 1. I think the entire `convertOrderBook` method should be placed inside fetchOrderBook and rewritten.